docs/apis/core/reportbuilder/index.md

```diff
@@ -0,0 +1,285 @@
+      new pix_icon('e/search', ''),
+      [],
+      true,
+      new lang_string('viewtasklog', 'report_tasklogs')
+  )));
+```
+
+### System reports
+
+System reports are a consistent way of providing reporting data, with paging, filtering, exporting standardized across them. Once the groundwork is done in defining the report elements in entities, it's possible to implement them with minimal code just by adding entities to the report, and defining which elements you want to use from them
+
+#### Create a new system report using entities
+
+To create a new system report just create a new class extending [reportbuilder/classes/system_report.php](https://github.com/moodle/moodle/blob/master/reportbuilder/classes/system_report.php).
+
+The first method that we need is ***initialise()*** :
+
+```php
+/**
+* Initialise report, we need to set the main table, load our entities and set columns/filters
+*/
+protected function initialise(): void {
+```
+
+The initialise method needs to get the main entity, set the main table it needs to use and add the entity to the report:
+
+```php
+// Our main entity, it contains all of the column definitions that we need.
+$entitymain = new task_log();
+$entitymainalias = $entitymain->get_table_alias('task_log');
+
+$this->set_main_table('task_log', $entitymainalias);
+$this->add_entity($entitymain);
+```
+
+After that, if the report will have 'Actions', it needs to define the columns these actions will use:
+
+```php
+$this->add_base_fields("{$entitymainalias}.id");
+```
+
+Now, after adding our first entity, the report can use the columns and filters from it OR more entities can be added to the report using SQL joins:
+
+```php
+$entityuser = new user();
+$entituseralias = $entityuser->get_table_alias('user');
+$this->add_entity($entityuser->add_join(
+    "LEFT JOIN {user} {$entituseralias} ON {$entituseralias}.id = {$entitymainalias}.userid"
+));
+```
+
+Once all entities have been added it needs to define which columns it needs to show **in the order we need**:
+
+```php
+$columns = [
+     'task_log:name',
+     'task_log:type',
+     'user:fullname',
+     'task_log:starttime',
+];
+
+$this->add_columns_from_entities($columns);
+```
+
+After defining the columns, it needs to define all the filters (or empty array for no filters) that it will use:
+
+```php
+$filters = [
+    'task_log:name',
+    'task_log:result',
+    'task_log:timestart',
+];
+
+$this->add_filters_from_entities($filters);
+```
+
+In case it needs actions for each report row, they can be defined like:
+
+```php
+// Action to download individual task log.
+$this->add_action((new action(
+    new moodle_url('/admin/tasklogs.php', ['logid' => ':id', 'download' => true]),
+    new pix_icon('t/download', ''),
+    [],
+    new lang_string('downloadtasklog', 'report_tasklogs')
+)));
+```
+
+**Note** that the placeholders used here (:id in this example) have been previously added using **add_base_fields**();
```

Comment:
```suggestion

:::info

Note that the placeholders used here (`:id` in this example) have been previously added using **add_base_fields**();

:::

```

Reply:
I have never used this one before, looks really nice! :clap: :smiley: